### PR TITLE
macOS wxWidgets 3.1.5 universal binary

### PR DESCRIPTION
**Pull request by sjanzou**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,7 +91,8 @@ function(set_default_compile_options target)
     else(MSVC)
         SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
         if (APPLE)
-            set(MAIN_CFLAGS "${MAIN_CFLAGS} -arch x86_64 -fno-common -DWX_PRECOMP -D__MACOSX__")
+#            set(MAIN_CFLAGS "${MAIN_CFLAGS} -arch x86_64 -fno-common -DWX_PRECOMP -D__MACOSX__")
+            set(MAIN_CFLAGS "${MAIN_CFLAGS} -fno-common -DWX_PRECOMP -D__MACOSX__")
```
Comment (dguittet) [inline CMakeLists.txt:95]: can you remove unused comment?

File: solarpilot/CMakeLists.txt [hunk 2]
```diff
@@ -68,7 +68,8 @@ if(MSVC)
 	endforeach()
 else()
 	if (APPLE)
-		add_compile_options(-arch x86_64 -fno-common)
+#       add_compile_options( -arch x86_64 -arch arm64  -fno-common)
+        add_compile_options( -fno-common)
```
Comment (dguittet) [inline solarpilot/CMakeLists.txt:72]: same here

File: ssc/CMakeLists.txt [hunk 3]
```diff
@@ -197,10 +197,10 @@ if(UNIX AND NOT SAMAPI_EXPORT)
...
 		target_link_libraries(dltest -ldl -Wl,--no-as-needed)
 	endif()
-	add_custom_command(
-		TARGET ssc
-		POST_BUILD
-		COMMAND dltest $<TARGET_FILE:ssc>)
+#	add_custom_command(
+#		TARGET ssc
+#		POST_BUILD
+#		COMMAND dltest $<TARGET_FILE:ssc>)
```
Comment (dguittet) [inline ssc/CMakeLists.txt:203]: why are we no longer running this?

**Review by dguittet — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (sjanzou) [reply CMakeLists.txt:95]: Will do... thanks!

File: ssc/CMakeLists.txt (hunk 3 above)
Comment (sjanzou) [reply ssc/CMakeLists.txt:203]: There was an issue with the build path, adding back now...

**Review by dguittet — APPROVED** [review]

File: ssc/CMakeLists.txt [hunk 4]
```diff
@@ -200,6 +200,7 @@ if(UNIX AND NOT SAMAPI_EXPORT)
 	add_custom_command(
 		TARGET ssc
 		POST_BUILD
+		COMMAND codesign --force --deep --sign - $<TARGET_FILE:ssc>
```
Comment (brtietz) [inline ssc/CMakeLists.txt:203]: This is causing me trouble on the Linux build - getting "codesign - not found" Is there a new dependency that should be added to the wiki?

**Review by brtietz — CHANGES_REQUESTED** [review]
Building with SAMAPI_EXPORT on works on linux, but off causes an error with the codesign command. (on fails the sam build due to the ssc library rename, which is expected)

**Comment by sjanzou** [comment]
That is an apple only thing.
Removing for Linux...thanks!

File: ssc/CMakeLists.txt (hunk 4 above)
Comment (sjanzou) [reply ssc/CMakeLists.txt]: Fixed in 8c9657eb7dab4d01be4c8faab7bd46ab96bfc4ce

**Review by brtietz — APPROVED** [review]
Testing looked good on Linux this morning. Nice work!

**Review by janinefreeman — APPROVED** [review]
Works for me on Windows with minimal testing

**Pull request merged**